### ISSUE TITLE
Can only finish current action error

### ISSUE THREAD

**Issue by TheCycoONE** [CorsixTH/CorsixTH#500]
_From [mark.l1...@hotmail.co.uk](https://code.google.com/u/117099726060447259318/) on September 12, 2012 14:49:59_

I came across this error, log, save and screen attached.

Latest SVN (downloaded tonight)and a new game.

Attempting to recover leaves you with two cardios and a stuck patient
Odd thing is if you load the save game you don't get the error, but the save was done just before the error occured.

I hope there is enough in the log to give you a clue
I have been running the game for about half an hour - I don't see that as being an issue though as it never has before.

**Attachment:** [gamelog.txt 1.png 10.sav](http://code.google.com/p/corsix-th/issues/detail?id=1290)

_Original issue: http://code.google.com/p/corsix-th/issues/detail?id=1290_

**Comment by bobcanthelpyou** [comment]
Game crash with the following message: 

```
Will not check for updates since this is the Trunk version.
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
./Lua/entities/humanoid.lua:662: Can only finish current action
stack traceback:
	[C]: in function 'assert'
	./Lua/entities/humanoid.lua:662: in function 'finishAction'
	./Lua/rooms/general_diag.lua:75: in function 'after_use'
	./Lua/humanoid_actions/multi_use_object.lua:264: in function 'timer_function'
	./Lua/entity.lua:218: in function 'tick'
	./Lua/world.lua:1108: in function 'onTick'
	./Lua/app.lua:876: in function 'dispatch'
	./Lua/app.lua:777: in function <./Lua/app.lua:772>

Warning: No event handler for timer
```

Savegame to reproduced (with game speed: 4): http://filebin.ca/25ENYOa9k5tE/Gamecrash.sav

**Comment by emilhem** [comment]
This issue is still occuring in 0.62
Updated output with new line numbers:
```
Lua\entities\humanoid.lua:618: Can only finish current action
stack traceback:
        [C]: in function 'assert'
        Lua\entities\humanoid.lua:618: in method 'finishAction'
        Lua\rooms\cardiogram.lua:91: in field 'after_use'
        Lua\humanoid_actions\multi_use_object.lua:316: in local 'timer_function'
        Lua\entity.lua:216: in method 'tick'
        Lua\world.lua:1036: in method 'onTick'
        Lua\app.lua:1014: in function <Lua\app.lua:1011>
        (...tail calls...)
        Lua\app.lua:915: in function <Lua\app.lua:910>
```

**Comment by mugmuggy** [comment]
Fixed with #1567
